tests/baselines/reference/bestCommonTypeOfTuple2.types

```diff
@@ -29,7 +29,7 @@ class F extends C { f }
 class C1 implements base1 { i = "foo"; c }
 >C1 : C1
 >base1 : base1
->i : string
+>i : any

```

Comment:
What do you plan to do when inferentially typing undefined or null? Will you leave it alone, to be later added as an inference candidate if it aligns with a type parameter? Or will you fix the type parameter, and then use that inference result to contextually type the undefined?